File: app/models/lesson.rb
class Lesson < ActiveRecord::Base
  belongs_to :day
  validates :title, presence: true, length: { maximum: 20 }
  validates :teacher, presence: true, length: { maximum: 20 }
  validates :time, presence: true, format: { with: /\d\d:\d\d/ }
end

File: app/assets/javascripts/models/lesson.js
Schedule.Lesson = DS.Model.extend({
  title: DS.attr('string'),
  teacher: DS.attr('string'),
  time: DS.attr('string'),
  day: DS.belongsTo('Schedule.Day'),

  becameInvalid: function () {
    location.reload(false);
  }
});